### PR TITLE
Add support for Bresser Lightning Sensor and Bresser Air Quality Sensor

### PR DESCRIPTION
**Pull request by matthias-bs**
See https://github.com/merbanan/rtl_433/issues/2140 and https://github.com/merbanan/rtl_433/issues/2693

### CODE REVIEW

**Comment by zuckschwerdt** [comment]
Yes, that's what I meant. But they should have the same r_device and  output_fields, otherwise this would give duplicate outputs.

**Comment by matthias-bs** [comment]
Do you mean I should merge both r_device and output_fields? How should I set the device name then?

**Comment by zuckschwerdt** [comment]
You can use a .name like you already sketched in the README: "Bresser Weather Center 7-in-1, Air Quality PM2.5/PM10"

**Comment by matthias-bs** [comment]
How about the assignment to `data`? Can I leave it as it is or do I have to move it out of the conditional code and assign dummy values for unavailable data?

**Comment by zuckschwerdt** [comment]
The `data` struct is flexible. I would even use
```
if (SENSOR_TYPE_WEATHER) { ... decoder_output_data(); return 1;}
else if (SENSOR_TYPE_AIR_PM) { ... decoder_output_data(); return 1;}
else { log... ; return DECODE_FAIL_SANITY;}
```

**Comment by matthias-bs** [comment]
All done now.

File: src/devices/bresser_7in1.c [hunk 1]
```diff
@@ -90,53 +107,86 @@ static int bresser_7in1_decode(r_device *decoder, bitbuffer_t *bitbuffer)
...
+                "humidity",         "Humidity",     DATA_INT,    humidity,
+                "wind_max_m_s",     "Wind Gust",    DATA_FORMAT, "%.1f m/s", DATA_DOUBLE, wgst_raw * 0.1f,
+                "wind_avg_m_s",     "Wind Speed",   DATA_FORMAT, "%.1f m/s", DATA_DOUBLE, wavg_raw * 0.1f,
+                "wind_dir_deg",     "Direction",    DATA_INT,    wdir,
+                "rain_mm",          "Rain",         DATA_FORMAT, "%.1f mm", DATA_DOUBLE, rain_mm,
+                "light_klx",        "Light",        DATA_FORMAT, "%.3f klx", DATA_DOUBLE, light_klx, // TODO: remove this
+                "light_lux",        "Light",        DATA_FORMAT, "%.3f lux", DATA_DOUBLE, light_lux,
+                "uv",               "UV Index",     DATA_FORMAT, "%.1f", DATA_DOUBLE, uv_index,
+                "battery_ok",       "Battery",      DATA_INT,    !battery_low,
+                "mic",              "Integrity",    DATA_STRING, "CRC",
```
Comment (zuckschwerdt) [inline src/devices/bresser_7in1.c:149]: `mic` must be last in all lists

File: src/devices/bresser_lightning.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",            "",                     DATA_STRING, "Bresser-Lightning",
+            "id",               "",                     DATA_FORMAT, "%08x",   DATA_INT,    sensor_id,
+            "battery_ok",       "Battery",              DATA_INT,    battery_ok,
+            "distance_km",      "storm_distance_km",    DATA_INT,    distance_km,
+            "strike_count",     "strike_count",         DATA_INT,    count,
+            "unknown1",         "Unknown1",             DATA_FORMAT, "%08x",   DATA_INT,    unknown1,
+            "unknown2",         "Unknown2",             DATA_FORMAT, "%08x",   DATA_INT,    unknown2,
+            "startup",          "Startup",              DATA_COND,   startup,  DATA_INT,    startup,
```
Comment (zuckschwerdt) [inline src/devices/bresser_lightning.c:105]: `mic` is missing

File: src/devices/bresser_lightning.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        return DECODE_ABORT_LENGTH; // message too short
+    }
+
+    bitbuffer_extract_bytes(bitbuffer, 0, start_pos, msg, sizeof(msg) * 8);
+
+    decoder_log_bitrow(decoder, 2, __func__, msg, sizeof(msg) * 8, "MSG");
+
+    int s_type      = msg[6] >> 4;
+    int chan        = msg[6] & 0x07;
+    int battery_ok  = (msg[5] & 0x08) ? 0 : 1;
```
Comment (zuckschwerdt) [inline src/devices/bresser_lightning.c:68]: we'd usually use a shift instead of a ternary, i.e. `>> 3` and then name it `battery_low`

**Comment by zuckschwerdt** [comment]
Looks almost perfect now, thanks!

File: src/devices/bresser_lightning.c (hunk 2 above)
Comment (matthias-bs) [reply src/devices/bresser_lightning.c]: added

File: src/devices/bresser_lightning.c (hunk 3 above)
Comment (matthias-bs) [reply src/devices/bresser_lightning.c]: changed, also for `startup` (renamed `nstartup`)

File: src/devices/bresser_7in1.c (hunk 1 above)
Comment (matthias-bs) [reply src/devices/bresser_7in1.c:150]: done

**Comment by zuckschwerdt** [comment]
Good for merge. Last call for comments ;)

**Comment by matthias-bs** [comment]
Done (for now)! Thank you!

**Pull request merged**